### PR TITLE
Results types

### PR DESCRIPTION
**Pull request by mikefranze**
## Description
Moves the tabulator results types to the domain model folder and adds the types to the front end so the results page can be fully typed.
Adds some generic types that can be extended for specific voting method types.

### CODE REVIEW

File: domain_model/ITabulators.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    pairwiseMatrix: pairwiseMatrix,
+}
+
+export interface allocatedScoreSummaryData extends starSummaryData {
+    splitPoints: number[],
+    spentAboves: number[],
+    weight_on_splits: number[],
+    weightedScoresByRound: number[][]
+}
+export interface approvalSummaryData extends genericSummaryData { }
```
Comment (ArendPeter) [inline domain_model/ITabulators.ts:55]: much cleaner :D

File: domain_model/ITabulators.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    summaryData: pluralitySummaryData,
+}
+
+export interface rankedRobinResults extends genericResults {
+    summaryData: rankedRobinSummaryData,
+}
+
+export interface irvResults extends Omit<genericResults, 'roundResults'> {
+    summaryData: rankedRobinSummaryData,
+    logs: string[],
```
Comment (ArendPeter) [inline domain_model/ITabulators.ts:104]: Not for this PR, but in the future should we remove logs in favor of putting it in roundResults?

**Review by ArendPeter — APPROVED** [review]
Looks great!

File: domain_model/ITabulators.ts (hunk 2 above)
Comment (mikefranze) [reply domain_model/ITabulators.ts:104]: Probably a good idea.

**Pull request merged**